src/metabase/api/dashboard.clj

```diff
@@ -234,16 +325,29 @@
                         :collection_id       collection_id
                         :collection_position collection_position
                         :is_app_page         (:is_app_page existing-dashboard)}
+        new-cards      (atom nil)
         dashboard      (db/transaction
-                         ;; Adding a new dashboard at `collection_position` could cause other dashboards in this
-                         ;; collection to change position, check that and fix up if needed
-                         (api/maybe-reconcile-collection-position! dashboard-data)
-                         ;; Ok, now save the Dashboard
-                         (u/prog1 (db/insert! Dashboard dashboard-data)
-                           ;; Get cards from existing dashboard and associate to copied dashboard
-                           (doseq [card (:ordered_cards existing-dashboard)]
-                             (api/check-500 (dashboard/add-dashcard! <> (:card_id card) card)))))]
+                        ;; Adding a new dashboard at `collection_position` could cause other dashboards in this
+                        ;; collection to change position, check that and fix up if needed
+                        (api/maybe-reconcile-collection-position! dashboard-data)
+                        ;; Ok, now save the Dashboard
+                        (let [dash (db/insert! Dashboard dashboard-data)
+                              id->new-card (when is_deep_copy
+                                             (duplicate-cards existing-dashboard collection_id))]
```

Comment:
nit: I think it'll be clearer if `duplicate-cards` return 2 keys `[:copied :uncopied]` rather than `[id1, id2,.. :uncopied]`

Reply:
done. good point.